Remove dockerinit once and for all

cyphar opened the pull request:
`dockerinit` has been in Docker since (AFAIK) the very beginning. It was once used by the LXC driver to set up containers. LXC has long since been removed and `dockerinit` has no real use anymore.

This could technically be included in `1.10`, but we should probably wait until after the release (although, it would be nice to remove LXC and the former LXC hack known as `dockerinit` in the same version).

Closes #18355

Signed-off-by: Aleksa Sarai asarai@suse.com

cyphar committed:
*: purge dockerinit from source code

dockerinit has been around for a very long time. It was originally used
as a way for us to do configuration for LXC containers once the
container had started. LXC is no longer supported, and /.dockerinit has
been dead code for quite a while. This removes all code and references
in code to dockerinit.

Signed-off-by: Aleksa Sarai <asarai@suse.com>

cyphar commented:
Okay, looks like everything is kosher (flaky tests aside).

cyphar committed:
*: remove documentation references to dockerinit

While the documentation is very patchy on dockerinit, remove all
references in packaging documentation to the now purged dockerinit.

Signed-off-by: Aleksa Sarai <asarai@suse.com>

cyphar commented:
This could technically be included in `1.10`, but we should probably wait until after the release (it would be nice to remove LXC and the former LXC hack known as `dockerinit`).

cyphar commented:
All the tests pass, so after the `engine-api` stuff is merged, we can merge this.

hqhq commented:
Glad to see we'll finally remove dockerinit, ping @tianon for reviewing.

tianon commented on hack/vendor.sh:
```diff
@@ -22,7 +22,7 @@ clone git github.com/vdemeester/shakers 3c10293ce22b900c27acad7b28656196fcc2f73b
 clone git github.com/docker/go-units 651fc226e7441360384da338d0fd37f2440ffbe3
 clone git github.com/docker/go-connections v0.1.2
-clone git github.com/docker/engine-api v0.2.2
+clone git github.com/docker/engine-api remove-dockerinit https://github.com/cyphar/engine-api
```
There's a PR open for this change, right? (On my phone now or I'd check directly)

tianon replied:
Aha https://github.com/docker/engine-api/issues/53 :+1:

tianon commented on hack/make/.go-autogen:
```diff
@@ -44,9 +41,9 @@ When make binary is run, the Dockerfile prepares the build environment by:
-   'go build', so no post-processing steps are required. The sources for 
+ - Run rsrc to create a binary file (autogen/winresources/rsrc.syso) that
+   contains the manifest and icon. This file is automatically picked up by
+   'go build', so no post-processing steps are required. The sources for
```
:heart:!

cyphar replied:
Haha, my `.vimrc` auto-removes trailing whitespace on `:w`.

jhowardmsft replied:
My bad putting the spaces in originally! :heart:

tianon commented:
I love it!

LK4D4 commented:
Very cool!

cyphar commented:
It looks like there's some outstanding issues with switching to `libseccomp-dev` instead of building `libseccomp`. I'll leave those for another PR.

tianon commented:
Yeah, that's fair IMO. :+1:

LK4D4 commented:
Yeah, I think it's enough for this PR :) Thanks @cyphar

cyphar commented:
@tianon @LK4D4 docker/engine-api#53 is waiting on this to move out of design review before merging.

LK4D4 commented:
@cyphar @calavera I moved it to code-review

estesp commented:
LGTM

thaJeztah commented:
@cyphar (and others here) w.r.t. the vendoring; we recently moved the "vendor" check to a separate check in CI; that allows you to include the changed vendor files as part of your PR, without making the tests fail (just the "vendor" check will fail; making it more apparent that we should not merge)

thaJeztah commented:
Is this something to be considered? https://github.com/docker/docker/issues/18355#issuecomment-161325606

cyphar commented:
@thaJeztah So you're saying I shouldn't update `vendor.sh` and just manually patch it?

thaJeztah commented:
@cyphar yes, I think that works; that we can review design/code changes, after that, the PR in the other repository can be merged, vendor.sh can be updated (and run) to get the "real" vendor changes in.

thaJeztah commented:
For now, this is fine, but I think that's a workflow that may work well (we're still getting used to the new repositories, so bear with us :smile:)

rhatdan commented:
@cyphar I was talking about things like hooks.d, which we would like to be able to execute from inside of docker containers.  We were looking at the location of dockerinit to figure out where to install a hook.  This is not a big deal, we will just move our hooks to be installed in /usr/lib/docker/ going forward, which seems to be the new standard for distributions, of where to launch helper applications.

cyphar commented:
The engine-api stuff has been merged, I've updated the PR.

/cc @tianon @hqhq @thaJeztah @LK4D4

LK4D4 commented:
@cyphar OMG, do you use some special go version or stuff? ping @tianon could you help pls?

cyphar commented:
@LK4D4 Oh god, it failed `vendor` too. I don't want to know what just happened. I'm re-vendoring on an Arch box.

cyphar commented:
Dammit `DefaultDockerHost` was broken upstream for `freebsd`.

calavera commented:
@cyphar I've opened https://github.com/docker/engine-api/pull/63 to revert the PullBehavior temporarily in engine-api. That way we'll be able to move forward with this change.

cyphar commented:
@calavera Are you sure that's necessary? My vendor patch implements the old behaviour (the tests pass AFAICS), so #16609 can actually implement the proper behaviour separately.

icecrime commented:
```
10:18:08 ---> Making bundle: validate-vendor (in bundles/1.10.0-dev/validate-vendor)
10:21:00 The result of ./hack/vendor.sh differs
10:21:00 
10:21:00  D vendor/src/github.com/docker/engine-api/client/client_nounix.go
10:21:00  D vendor/src/github.com/docker/engine-api/client/client_unix.go
10:21:00 ?? vendor/src/github.com/docker/engine-api/client/client_darwin.go
10:21:00 ?? vendor/src/github.com/docker/engine-api/client/client_linux.go
10:21:00 ?? vendor/src/github.com/docker/engine-api/client/client_windows.go
``` :sob:

jhowardmsft commented:
@icecrime Can you re-run CI (not on IRC). I fixed this earlier this afternoon

cyphar commented:
@icecrime Yeah, that's intentional (it was for testing while I waited for the engine-api stuff to be merged, and apparently we're not meant to commit `hack/vendor.sh` changes for unmerged patches?). I'll fix this up tomorrow.

vdemeester commented:
@cyphar yep, that's a way to validate that changes in vendor are not done manually / are the same as upstream. It's on a separate build to make the *normal* builds be ok but to warn maintainers that there is still some work to do on the vendoring (like waiting for it to be merged upstream) :stuck_out_tongue:.

cyphar committed:
vendor: update engine-api

This removes all references to InitPath and InitSha1, as well as pulling
in a few other minor engine-api fixes.

Signed-off-by: Aleksa Sarai <asarai@suse.com>

cyphar commented:
I've rebased and updated to use the latest engine-api (hopefully that fixes all of our problems).

/cc @tianon @LK4D4 @estesp @calavera @icecrime @vdemeester 

@LK4D4 @tianon: Unrelated to this PR, it looks like my openSUSE machine is breaking `hack/vendor.sh` (cleanup removes files it shouldn't remove). I'll look into this later.

estesp commented:
still LGTM

runcom commented:
LGTM

vdemeester commented:
LGTM 🐯
/ping @docker/core-maintainers :smile_cat:

tianon commented:
Nice, LGTM

icecrime commented:
LGTM :heartbeat:

LK4D4 commented:
OMG, congrats all!

jfrazelle commented:
yayyyyy! thanks @cyphar

On Tue, Jan 26, 2016 at 8:30 AM, Arnaud Porterie <notifications@github.com>
wrote:

> Merged #19490 <https://github.com/docker/docker/pull/19490>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/19490#event-526729126>.
>

cyphar commented:
w00t w00t!

tianon commented:
This is so good :heart:

vdemeester commented:
:tada:
